### PR TITLE
proper pull request

### PR DESCRIPTION
**Pull request by TacoTheDank**
pretty much the same as before but a lot better code review because done in android studio

also properly updated gradles and gradlews by using ./gradlew wrapper --gradle-version 4.9 --distribution-type all

android studio cleans and builds successfully

heres apks built from this
https://app.box.com/v/redditandroidappstore

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -3,14 +3,14 @@ apply plugin: 'realm-android'
 apply plugin: 'com.github.kt3k.coveralls'
 apply plugin: 'jacoco'
 
-def gitSha() {
+/*def gitSha() {
```
Comment (d4rken) [inline app/build.gradle]: Why did you comment this out?

File: app/src/main/java/subreddit/android/appstore/AndroidModule.java [hunk 2]
```diff
@@ -8,12 +8,13 @@
 import dagger.Provides;
 import subreddit.android.appstore.util.dagger.ApplicationScope;
 
-
 @Module
 public class AndroidModule {
     private final AppStoreApp app;
 
-    public AndroidModule(AppStoreApp app) {this.app = app;}
+    AndroidModule(AppStoreApp app) {
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/AndroidModule.java]: Don't just change formatting for the sake of it. While there is no style guide attached to this project, just changing formatting is usually not worth a commit.

File: app/build.gradle [hunk 3]
```diff
@@ -96,10 +95,12 @@ android {
             outputs.upToDateWhen { false }
             showStandardStreams = true
         }
-
         jacoco {
             includeNoLocationClasses = true
         }
+        tasks.withType(Test) {
```
Comment (d4rken) [inline app/build.gradle]: Why did you add this?

Either comment in code, or in the commit message why certain changes have been made.

File: app/src/androidTest/java/subreddit/android/appstore/ApplicationTest.java [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
 /**
- * <a href="http://d.android.com/tools/testing/testing_android.html">Testing Fundamentals</a>
+ * Instrumented test, which will execute on an Android device.
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
  */
-public class ApplicationTest extends ApplicationTestCase<Application> {
-    public ApplicationTest() {
-        super(Application.class);
+
+@RunWith(AndroidJUnit4.class)
```
Comment (d4rken) [inline app/src/androidTest/java/subreddit/android/appstore/ApplicationTest.java]: Fixing deprecated testing for newer build tools is good, this would perfectly fit a sepperate PR that just updates testing 👍

File: app/src/main/java/subreddit/android/appstore/AppStoreApp.java [hunk 5]
```diff
@@ -81,6 +81,7 @@ public void restart() {
         theme = Integer.parseInt(PreferenceManager.getDefaultSharedPreferences(this).getString("theme", "0"));
         Intent i = getBaseContext().getPackageManager()
                 .getLaunchIntentForPackage(getBaseContext().getPackageName());
+        assert i != null;
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/AppStoreApp.java]: `assert` doesn't work in production builds.

e.g. https://stackoverflow.com/questions/6176441/how-to-use-assert-in-android

Generally some NPE warnings are okay, because think about what would happen if, `i` is `null`.

If we check and throw, we just throw an exception one line earlier at the cost of adding an extra line of code. Adding an explicitly check is certainly not bad, but what do we gain? Sometime (in my opinion) it's okay to let code run into unchecked NPE`s if it is easily noticeable where the NPE came from.

File: app/src/main/java/subreddit/android/appstore/backend/DeviceIdentifier.java [hunk 6]
```diff
@@ -21,5 +21,4 @@ public String getUUID() {
         }
         return uuid;
     }
-
-}
+}
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/backend/DeviceIdentifier.java]: I'd merge changes like this, though I don't consider them necessary. Such formatting changes should occur in a single PR that does not change code behavior at all, but just formatting.

File: app/src/main/java/subreddit/android/appstore/backend/data/AppInfo.java [hunk 7]
```diff
@@ -65,25 +72,16 @@ public String getDescription() {
         return appTagsCollection;
     }
 
-    public Collection<AppTags> addTag(AppTags tag) {
-        if(!appTagsCollection.contains(tag)){
+    public void addTag(AppTags tag) {
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/backend/data/AppInfo.java]: Not a fan of PRs that reorder methods, I've never seen any other project that accepts PR on reordering methods either.

Did you just execute all code reformating options? It's usually best practice to only reformat code that you yourself touched, trying to touch as few other files as possible that are unrelated to ones changes.

File: app/src/main/java/subreddit/android/appstore/backend/scrapers/LiveMediaScraper.java [hunk 8]
```diff
@@ -14,9 +13,9 @@
 import timber.log.Timber;
 
 public class LiveMediaScraper implements MediaScraper {
-    final LruCache<AppInfo, Observable<ScrapeResult>> scrapeCache = new LruCache<>(1);
-    final ScrapeDiskCache scrapeDiskCache;
-    final GPlayScraper gPlayScraper;
+    private final LruCache<AppInfo, Observable<ScrapeResult>> scrapeCache = new LruCache<>(1);
+    private final ScrapeDiskCache scrapeDiskCache;
+    private final GPlayScraper gPlayScraper;
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/backend/scrapers/LiveMediaScraper.java]: I currently can't check via AS but I think they were class protected on purpose.

Either due to testing access or because of synthetic accessors.

https://www.google.de/search?q=synthetic+accessors+android

File: app/src/main/java/subreddit/android/appstore/backend/scrapers/gplay/GPlayScraper.java [hunk 9]
```diff
@@ -48,24 +48,24 @@ public GPlayScraper(OkHttpClient client) {
                 .map(response -> {
                     if (response.code() == 404) throw new DeadLinkException(response.request().url().toString());
 
-                    Document doc = Jsoup.parse(response.body().string());
+                    Document doc = Jsoup.parse(Objects.requireNonNull(response.body()).string());
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/backend/scrapers/gplay/GPlayScraper.java]: The app is minAPI 16. AFAIK `requireNonNull` is API 19. So this would should crash, did you test run?

**Comment by TacoTheDank** [comment]
i must look so desperate lmao, god i suck

welp time to do it again xd

**Comment by d4rken** [comment]
> i must look so desperate lmao, god i suck

Not at all, I admire your efforts.

> welp time to fix it xd

Just go with small PRs.
